app/controllers/notes_controller.rb

```diff
@@ -461,4 +448,15 @@ def location
 
     [latitude, longitude]
   end
+
+  def published_notes
+    hidden_nids = Node.hidden_response_node_ids
+    notes = Node.research_notes.where('node.status = 1').order('node.nid DESC')
```

Comment:
@RuthNjeri I think it could be its because you are already doing an order by nid on `published_notes` on line 454, maybe we could remove the order here and just do it on the methods...what do you think?

Reply:
Hey @cesswairimu in this [version](https://github.com/publiclab/plots2/pull/8800#issuecomment-742583719) (the screenshot), I did that, I only pushed the current one (line 454) after I tested it.. the results are the same.. 

I've attached other screenshots..

<img width="784" alt="Screenshot 2020-12-10 at 20 41 46" src="https://user-images.githubusercontent.com/7622875/101809151-40fd3a00-3b28-11eb-82f1-abb96a53d050.png">

<img width="597" alt="Screenshot 2020-12-10 at 20 42 05" src="https://user-images.githubusercontent.com/7622875/101809161-43f82a80-3b28-11eb-8db1-321fb2c2bbb6.png">

![Screenshot 2020-12-10 at 20 42 13](https://user-images.githubusercontent.com/7622875/101809218-54100a00-3b28-11eb-8b14-cefa4ab2319b.png)

Reply:
Maybe the node.nid order takes precedence?

Reply:
I believe what the prob is the double `.order`...we could try (with your `published_notes` method as the above screenshot )
```
def liked
    @title = I18n.t('notes_controller.highly_liked_research_notes')
    @notes = published_notes
             .order('cached_likes DESC').limit 20
    @unpaginated = true
    render template: 'notes/index'
  end
```
I don't think we need to order by latest on most liked...then you can have the order on `recent` what do you think?

Reply:
Also just seen this for multiple ordering...maybe you could try it out `.order(cached_likes: :desc, nid: :desc)`

Reply:
Thanks @cesswairimu let me run it in the debugger and share the results

Reply:
For this  `I don't think we need to order by latest on most liked...then you can have the order by nid on recent what do you think?` It makes sense, though with the limit of 20 nodes, I think it defaults in picking the earliest 20 nodes...

Reply:
@cesswairimu the query you shared works as expected 😄 , I think the issue was with the way the arguments within the order params were positioned.. in the first query it was
`Node.research_notes.where('node.status = 1').order('node.nid DESC').order('cached_likes DESC')`
The node.nid order is positioned first so the default order will be according to the` node.nid DESC`

And the second one that you've suggested `Node.research_notes.where('node.status = 1').order(cached_likes: :desc, nid: :desc)`
The node.nid order is positioned second.. and it prioritizes the cached_likes

<img width="925" alt="Screenshot 2020-12-10 at 22 09 27" src="https://user-images.githubusercontent.com/7622875/101818060-6132f600-3b34-11eb-8966-80b9d81f03b7.png">

(the order is based on the most recent with the highest number of likes)


I've tested it with the same query you shared but switched up the position and got the same original problem,
`Node.research_notes.where('node.status = 1').order(nid: :desc, cached_likes: :desc, )`

<img width="888" alt="Screenshot 2020-12-10 at 22 11 26" src="https://user-images.githubusercontent.com/7622875/101818268-abb47280-3b34-11eb-9ee1-c210cf0e5189.png">

(the order is based on the most recent and not exactly on the highest number of likes)... still confusing to me but so far this is the only thing that makes sense..


It's now fixed 🎉 , Thanks for the help @cesswairimu 😄